### DISCUSSION TITLE
How to test federation? (Dev Environment)

### DISCUSSION THREAD

**Discussion by kartonrad** [mastodon/mastodon#21331] [Q&A]
I have a dev enviroment in wsl2...
how do i set up a second instance to talk to it, without too much headache?

is there a tool to simulate an activity being delivered?

i tried to use vagrant, but it couldn't run foreman start outside of the home directory
i tried literally everything (it's been hours and hours but no luck)
(setting PATH, running from home dir with various command line arguments, editing the Procfile)

i am on windows with virtual box, but honestly, i'm over vagrant, it just doesn't work rn

i already have a working dev instance, i hust need something to federate with (moving followers needs to be supported)
Yall know anything?

**Comment by bobhy** [comment]
have you tried running a second dev environment in a second instance of WSL?  https://stackoverflow.com/questions/51584765/how-do-you-install-multiple-separate-instances-of-ubuntu-in-wsl
